### PR TITLE
Added log when currency rates import fail

### PR DESCRIPTION
**Pull request by kyrena**
### Description

When currency cron "Scheduled Import Settings" is enabled, in System > Configuration > Currency Setup, when there is an import error, errors are not logged.

With this PR, error is logged in exception.log (`Mage::logException`), cron job is marked as "error" (`$cron->setIsError`), and cron message contains the error log.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All automated tests passed successfully (all builds are green)
 - [ ] Add yourself to contributors list

### CODE REVIEW

File: app/code/core/Mage/Directory/Model/Observer.php [hunk 1]
```diff
@@ -22,49 +22,57 @@
...
+    public const XML_PATH_ERROR_TEMPLATE  = 'currency/import/error_email_template';
+    public const XML_PATH_ERROR_IDENTITY  = 'currency/import/error_email_identity';
     public const XML_PATH_ERROR_RECIPIENT = 'currency/import/error_email';
 
     /**
+     * @param Mage_Cron_Model_Schedule|null $cron
      * @throws Mage_Core_Exception
      */
-    public function scheduledUpdateCurrencyRates()
+    public function scheduledUpdateCurrencyRates($cron = null)
```
Comment (sreichel) [inline app/code/core/Mage/Directory/Model/Observer.php:36]: Pls dont signature of used methods. Better implement a new one,

Comment (kiatng) [reply app/code/core/Mage/Directory/Model/Observer.php:36]: How about this?
```php
if (func_num_args() && func_get_args()[0] instanceof Mage_Cron_Model_Schedule) {
    $schedule = func_get_args()[0];
    //....
}

**Comment by kiatng** [comment]
For BC, it's tricky to remove entries in the translation file `Mage_Directory.csv`.

File: app/code/core/Mage/Directory/Model/Observer.php (hunk 1 above)
Comment (sreichel) [reply app/code/core/Mage/Directory/Model/Observer.php:36]: I'd add a new method w/ parameter and all logic in and forward to this method. Similiar i did with #2940.

**Comment by fballiano** [comment]
Since the original author didn't answer any of the notes I'll close this PR.
@kyrena feel free to reopen if you're willing to fix it :-)

**Pull request closed without merging**